### DISCUSSION TITLE
Setting password after sign in with oauth

### DISCUSSION THREAD

**Discussion by alexander-densley** [supabase/supabase#26395] [Questions]
In my set up right now, if a user sign's up with email and password, and then later uses sign in with google, the providers are linked automatically. However, if a user sign's up with oauth first and then later "sets a password", the password does add to their user data, and they can sign in, but email is not listed in the providers. To be fair, my current approach to adding the password post oauth signup was by using the "updateUser", I imagine this might be part of the problem? Would appreciate any advice, thanks

**Comment by Glorfindel31** [comment]
**I have here some similar problems:**

- We are unable to open the OAuth callback in a new window.
- Check if the user exists.
- In the case the account was used before, then the automatic behavior is to log in instead of re-getting or throwing errors.

**Comment by mobidroid92** [comment]
I went into the same problem, the only solution is to make a stored procedure function to store the new identity (email) into the auth.identity table, then call it as rpc (remote procedure call) inside an edge function.
